src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let string = '';
+
+    for (let value = 0; value < this.length; value++) {
```

Comment:
```suggestion
    for (let i = 0; value < this.length; value++) {
    const value = this[i];
```
but `for of`  will better here